autoload/neoterm/repl/python.vim

```diff
@@ -0,0 +1,67 @@
+"init python specific settings
+if !exists('g:neoterm_repl_ipy_magic')
+  let g:neoterm_repl_ipy_magic = 0
+end
+
+function! neoterm#repl#python#is_valid(value)
+  let l:pyCmd = 0
+  let l:invalid = 0
+  if type(a:value) == v:t_list
+    for i in range(len(a:value))
+      if a:value[i] =~ 'python'
+        let l:pyCmd = 1
+        if a:value[i] == 'ipython'
+          let g:neoterm_repl_python[i] = 'ipython --no-autoindent'
+        endif
+      endif
+      if executable(split(a:value[i])[0]) == 0
+        let l:invalid = 1
+      endif
+    endfor
+  else
+    if a:value =~ 'python'
+      let l:pyCmd = 1
+      if a:value == 'ipython'
+       let g:neoterm_repl_python = 'ipython --no-autoindent'
+    endif
+    endif
+    if executable(split(a:value)[0]) == 0
+      let l:invalid = 1
+    endif
+  endif
+  if l:pyCmd == 1 && l:invalid == 0
+      return 1
+  else
+      return 0
+  endif
+endfunction
+
+function! neoterm#repl#python#exec(command)
```

Comment:
What do you think to make things a bit more explicit in here, as well, like:
```vimscript
function! neoterm#repl#python#exec(command)
  if g:neoterm_repl_ipy_magic == 1 && join(g:neoterm_repl_command) =~ "ipython"
    let l:cmd = s:ipython_magic_command_for(a:command)
  elseif join(g:neoterm_repl_command) !~ 'ipython'
    let l:cmd = s:ipython_command_for(a:command)
  else
    let l:cmd = s:python_command_for(a:command)
  end

  call g:neoterm.repl.instance().exec(add(l:cmd, g:neoterm_eof))
endfunction
```

This way we'd have one private function for each way neoterm can send python commands to the REPL, it'd be easier to debug and easier to maintain. 😄

Reply:
Yeah, of course.
How do you debug vim scripts?

Reply:
When needed, yes, but usually only with `echo` 🙈